fixit/rules/chained_instance_check.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Set
+
+import libcst as cst
+import libcst.matchers as m
+
+from fixit import (
+    CstContext,
+    CstLintRule,
+    InvalidTestCase as Invalid,
+    ValidTestCase as Valid,
+)
+
+
+class ChainedInstanceCheckRule(CstLintRule):
+    """
+    The built-in ``isinstance`` function, instead of a single type,
+    can take a tuple of types and check whether given target suits
+    any of them. Rather than chaining multiple ``isinstance`` calls
+    with a boolean-or operation, a single ``isinstance`` call where
+    the second argument is a tuple of all types can be used.
+    """
+
+    MESSAGE: str = (
+        "Multiple isinstance calls with the same target but "
+        + "different types can be collapsed into a single call "
+        + "with a tuple of types."
+    )
+
+    VALID = [
+        Valid("foo(x, y) or foo(x, z)"),
+        Valid("foo(x, y) or foo(x, z) or foo(x, q)"),
+        Valid("isinstance(x) or isinstance(x)"),
+        Valid("isinstance(x, y) or isinstance(x)"),
+        Valid("isinstance(x) or isinstance(x, y)"),
+        Valid("isinstance(x, y) or isinstance(t, y)"),
+        Valid("isinstance(x, y) and isinstance(x, z)"),
+        Valid("isinstance(x, y) or isinstance(x, (z, q))"),
+        Valid("isinstance(x, (y, z)) or isinstance(x, q)"),
```

Comment:
In my opinion, no. Since the developer is aware that they can use a tuple of types in `isinstance` calls, the best estimation would be that they wanted to separate a check into two different groups for readability.
```py
if isinstance(x, (int, float, complex)) or isinstance(x, DateT): # either numeric or date
     ...
```

Reply:
Fair enough.